### DISCUSSION TITLE
Welcome to FusionCache Discussions!

### DISCUSSION THREAD

**Discussion by jodydonetti** [ZiggyCreatures/FusionCache#54] [Announcements]
## 👋 Welcome!

We’re using Discussions as a place to connect with other members of our community.

We hope that you:
* 🤔 Ask questions you’re wondering about
* 🤪 Share some ideas, even the wild ones!
* 💃 Engage with other community members and try to be helpful

Some of the existing issues have been converted to discussions, so we can talk about them and see where they go.
If they end up generating an actual new feature or change, an issue will be created to track design & development.

**Comment by AhmedSameh2020** [comment]
Hi Jody,
I have a question related to unit testing, I am trying to mock some methods like the following but I get NullReferenceException.

```
   _cacheMock.Setup(x => x.ExpireAsync(It.IsAny<string>(), It.IsAny<FusionCacheEntryOptions>(), It.IsAny<CancellationToken>()));
```
```
  Message: 
    Did not expect any exception, but found System.NullReferenceException: Object reference not set to an instance of an object.
       at ZiggyCreatures.Caching.Fusion.FusionCacheExtMethods.ExpireAsync(IFusionCache cache, String key, Action`1 setupAction, CancellationToken token) in /_/src/ZiggyCreatures.FusionCache/FusionCacheExtMethods.cs:line 279
```
       
       Can you please tell me what I am doing wrong?

**Reply by jodydonetti** [reply]
Hi @AhmedSameh2020 , sorry but I don't have enough information to answer you.

One thing I can tell you though is that what you are trying to execute inside the `.Setup(...)` method call (meaning that `ExpireAsync(...)` overload) is an extension method and not an instance method, maybe this can help?

ps: if you can, please create a separate discussion, thanks!

**Comment by Xela3011** [comment]
Hi, Jody. I also have a question related to unit testing. I'm trying to mock a FusionCache call using FakeItEasy in the following way:
```
private readonly IFusionCache cache;
 cache = A.Fake<IFusionCache>(); (this goes inside constructor)
 A.CallTo(() => cache.GetOrSet("ratesUSD", A<Func<FusionCacheEntryOptions, decimal>>.Ignored, null, null, default)).Returns(call => 56.60m);
  var result = await controller.GetRate("USD");
```

It doesn't give me any exception, but when i call the GetOrSet method in the action, doesn't return anything:

```
        public async Task<JsonResult> GetRate(string currency) => await Task.Run(() =>
        {
            var data = cache.GetOrSet($"rates{currency}", _ => classificationCompanyService.GetRate(currency));
            return Json(new { data });
        });

````

could you point me in the right direction to perform this mock, please?

**Reply by jodydonetti** [reply]
Hi @Xela3011 , sorry for the delay, it's a busy period.

Have you been able to solve this meanwhile?
